File: CSharp/Teste3D/Teste3D/Pixel.cs
namespace Teste3D
{
    public class Pixel
    {
        public float nX, nY = 0;
        public int nCor     = 0;

        public Pixel()
        {
        }

        public Pixel(int tnCor, int tnX,int tnY)
        {
            nCor = tnCor;
            nX  = tnX;
            nY  = tnY;
        }

        public Pixel(int tnCor, float tnX, float tnY)
        {
            nCor = tnCor;
            nX = tnX;
            nY = tnY;
        }
    }
}

File: Java/JJPaint/src/Classes/Pixel.java
package Classes;


import java.awt.Color;

/**
 *
 * @author rmendonca
 */
public class Pixel {
    //Propriedades da Classe
    public int nX       = 0;
    public int nY       = 0;
    public Color oColor = new Color(0);

    public Pixel(){
    
    }
    public Pixel(int tnX,int tnY){
        nX = tnX;
        nY = tnY;
    }
    public Pixel(Color toColor){
        oColor = toColor;
    }
    public Pixel(int tnX,int tnY,Color toColor){
        nX = tnX;
        nY = tnY;
        oColor = toColor;
    }

    public static Pixel[] Fusion(Pixel[]... toPixel){
        Pixel[] oFusion = null;
        int nTam = 0,nJ = 0;
        //faz a contragem dos tamanhos dos vetores
        for (Pixel[] oPi : toPixel) {
            nTam+= oPi.length;
        }
        oFusion = new Pixel[nTam];
        for (Pixel[] oPi : toPixel) {
            for (int nI = 0; nI < oPi.length; nI++) {
                oFusion[nJ] = oPi[nI];
                nJ++;
            }
        }
        return(oFusion);
    }
}
